### ISSUE TITLE
Motivating example

### ISSUE THREAD

**Issue by zoosky** [jducoeur/gadgets#2]
I've created a project with the motivation example:
https://github.com/zoosky/gadgets-example

Unfortunately  some Types are missing: Date, TransactionList, DatePickerGadget.

Any hints?

**Comment by jducoeur** [comment]
Ah -- sorry, that README was written in haste, because I was pulling the library out of Querki so I could use it for a client, and that section is unclear.  It is basically just an example of what the middle of a hypothetical stack might look like, to show the way the pieces compose, but it's missing lots of stuff, and wasn't intended to be compileable in its current state.  I've tweaked the wording to hopefully make that clearer.

In practice, I grabbed that example because it was a piece of what I was planning to write for said client, so it was top-of-mind for me.  I can't copy the actual code up to here (since it is proprietary), so I'll probably replace that example with something more fully-worked when I get an hour to do so.  I'm going to leave this Issue open as a reminder to do that.

But in the meantime, to understand the example, figure that `Date` means the JavaScript Date type, `DatePickerGadget` is a Gadget wrapping around the [bootstrap-datepicker](https://github.com/jducoeur/bootstrap-datepicker-scalajs), and `TransactionList` is another Gadget in this stack, that fetches transactions from the server based on the selected dates, and displays them.

For more examples in the meantime, I encourage you to poke around [Querki](https://github.com/jducoeur/Querki/tree/feature/update-apps), the project from which this library was extracted -- if you search in there, you'll find dozens of Gadgets in the client code.
